warehouse/macaroons/caveats/__init__.py

```diff
@@ -105,6 +106,7 @@ def verify(self, request: Request, context: Any, permission: str) -> Result:
 @dataclass(frozen=True)
 class OIDCPublisher(Caveat):
     oidc_publisher_id: StrictStr
+    oidc_claims: SignedClaims | None = None
```

Comment:
Embedding the whole `SignedClaims` is going to make this macaroon very large, and they're already very large.

Instead of doing the whole thing, we should probably limit it to the set of claims needed for the events we plan to emit. This will additionally prevent any unintentional forwarding of "secret" claims (there shouldn't be any currently, but future OIDC IdPs may choose to embed claims that shouldn't be forwarded).